lgc/patch/PatchBufferOp.cpp

```diff
@@ -1579,17 +1577,12 @@ Value *PatchBufferOp::replaceICmp(ICmpInst *const iCmpInst) {
 
   assert(iCmpInst->getPredicate() == ICmpInst::ICMP_EQ || iCmpInst->getPredicate() == ICmpInst::ICMP_NE);
 
-  Value *bufferDescICmp = m_builder->getFalse();
-  if (!bufferDescs[0] && !bufferDescs[1])
-    bufferDescICmp = m_builder->getTrue();
-  else if (bufferDescs[0] && bufferDescs[1]) {
-    Value *const bufferDescEqual = m_builder->CreateICmpEQ(bufferDescs[0], bufferDescs[1]);
+  Value *const bufferDescEqual = m_builder->CreateICmpEQ(bufferDescs[0], bufferDescs[1]);
 
-    bufferDescICmp = m_builder->CreateExtractElement(bufferDescEqual, static_cast<uint64_t>(0));
-    for (unsigned i = 1; i < 4; ++i) {
-      Value *bufferDescElemEqual = m_builder->CreateExtractElement(bufferDescEqual, i);
-      bufferDescICmp = m_builder->CreateAnd(bufferDescICmp, bufferDescElemEqual);
-    }
+  Value *bufferDescICmp = m_builder->CreateExtractElement(bufferDescEqual, static_cast<uint64_t>(0));
```

Comment:
Is the change here to no longer handle null directly, but rely on the newly generated real null buffer descriptor?